### ISSUE TITLE
Code Coverage: Create tests for OrgSettings.tsx

### ISSUE THREAD

**Issue by palisadoes** [PalisadoesFoundation/talawa-admin#396]
- Please coordinate **issue assignment** and **PR reviews** with the contributors listed in this issue https://github.com/PalisadoesFoundation/talawa/issues/359

We will need unittests done for all methods, classes and functions found in this file.

Any widgets referenced in this file must also have unittests done

```
src/screens/OrgSettings/OrgSettings.tsx
```

- Parent Issue: https://github.com/PalisadoesFoundation/talawa-admin/issues/241

### PR Acceptance Criteria

- When complete this file must show **100%** coverage when merged into the code base.
- [The current code coverage for the file can be found here](https://app.codecov.io/gh/PalisadoesFoundation/talawa-admin?search=&displayType=list)
- The PR will show a report for the code coverage for the file you have added. You can use that as a guide

**Comment by saadabban76** [comment]
what does create tests mean ? sorry if this is a silly question, I'm new to the open source contribution :)

**Comment by palisadoes** [comment]
Please research typescript testing. It's a necessary skill to ensure reliability in any programming language

**Comment by noman2002** [comment]
@saadabban76 Its a genuine question don't worry about it. So basically writing tests means writing unittest for various functionality and UI to make the code reliable. Here, this issue means to test the code present orgSetting.ts file by writing unit tests. Hopefully this answers your question. You can also join the slack channel for detailed conversation and to communicate with other members. Thank you.

**Comment by saadabban76** [comment]
@noman2002 thanks for the descriptive reply, I understood this 👍, also can you please share the slack channel link to join ?

**Comment by noman2002** [comment]
You can find it in the readme.me of talawa repository.

**Comment by anurag0006** [comment]
As no one is working on this, i Would like to work on this issue. Please assign it to me.

**Comment by anurag0006** [comment]
@palisadoes ,please assign this issue to me

**Comment by Ayush-sss** [comment]
I want to work on this issue. Please assign this issue to me and allow me to work on it.

**Comment by anurag0006** [comment]
@Ayush-sss, i am already working on this issue...Please find some other issue to work on...There are some other unassigned issues in the issue section, u can go there and comment under the issue for it to be assigned to u

**Comment by anurag0006** [comment]
@palisadoes sir, i was working on this issue and i am just left with one line of code which is not under the code coverage...I tried to reverse engineer the testing code from other files too but wasn't able to find anything useful...I have already messaged about it on the slack(talawa-admin) in full detail about the line which am not able to cover and asked for help but no one responded...Can you allot me a mentor or someone which can help me out to complete the code coverage for this issue?

**Comment by palisadoes** [comment]
1. Please submit the PR with that explanation. We can merge if approved and reopen this issue for someone else to figure out the fix.
1. Keep a track of this issue to see how this could be overcome.

We are all learning.

**Comment by anurag0006** [comment]
@palisadoes sir, I have already added added those test cases in my other PR in which i solved a UI issue...Earlier the code coverage for this file was about 40% but now its around 89%...I thought i would make a PR for this one when i achieved 100% coverage but am facing the above mentioned issue....So should i reopen the issue with full explanation?

**Comment by palisadoes** [comment]
Are you saying that the testing that final line of code will move you from 89% to 100%?

**Comment by anurag0006** [comment]
In my terminal, when i am testing using "yarn test --coverage" , it shows that only one line is not under coverage 
![image](https://user-images.githubusercontent.com/107215525/218282979-9ee8147f-6af7-411b-96f1-c77a25e8df07.png)

but when i see the same in the codecov, it shows that the 3 lines are not under coverage as showen in the attached screenshot( the lines marked with a red slash)
![image](https://user-images.githubusercontent.com/107215525/218283012-4fc70e4d-f18e-48ae-b064-1d63b132de0b.png)



> Are you saying that the testing that final line of code will move you from 89% to 100%?
Yeah if go through the terminal coverage..but even if we see the coverage shown by codecov, only 3 lines are out of coverage and all 3 are interrelated so , its just one small block of code which is not under coverage

**Comment by palisadoes** [comment]
Don't spend to much time in this. Unassign yourself and let someone else give it a try.

You are reaching diminishing returns.

**Comment by kb-0311** [comment]
can I be assgined this issue instead ?

**Comment by noman2002** [comment]
@kb-0311 Any update ?

**Comment by bigBrain1901** [comment]
Hey! I would love to give a shot at this issue. Can I be assigned, please?

**Comment by bigBrain1901** [comment]
Hey! I would love to give a shot at this issue. Can I be assigned, please?

**Comment by bigBrain1901** [comment]
## Introduction
I have spotted an issue with the following line of code.  
[/src/screens/OrgSettings/OrgSettings.tsx#L142](https://github.com/PalisadoesFoundation/talawa-admin/blame/1fa83d56569804b4dc87a759fdc612f127c7de6e/src/screens/OrgSettings/OrgSettings.tsx#L142).

This is the line.
```js
data.organizations.membershipRequests
```

## Explanation of the Issue

On lines [L46-L48](https://github.com/PalisadoesFoundation/talawa-admin/blame/1fa83d56569804b4dc87a759fdc612f127c7de6e/src/screens/OrgSettings/OrgSettings.tsx#L46) in the same file, the following query is made on the organization ID from the URL.
```js
const { data, loading, error } = useQuery(MEMBERSHIP_REQUEST, {
    variables: { id: currentUrl },
});
```

The object data has the following structure.
```json
{
    "organizations": [
        {
            "_id": "id",
            "membershipRequests": [
                {
                    "_id": string,
                    "user": {
                        "_id": string,
                        "firstName": string,
                        "lastName": string,
                        "email": string,
                    },
                },
                // other such request objects
            ]
        },
    ]
}
```

It is noteworthy that the number of `Organization` objects in the `data.organizations` will always be 1, as the query for the `membershipRequests` is made on the `organisation ID` as shown above.

Hence, going back to the line L142 in the [Introduction](## Introduction) section above, it is indeed the case that this line is semantically wrong.

## Possible Fixes

There are 2 possible fixes to this issue, that I can see

1. The correct condition must be set to be the following.
```js
data.organizations[0].membershipRequests.length > 0
```

2. Edit the `talawa-api` API to send back the response as a simple array of `membershipRequests` rather than this complex data structure. The API endpoint could generate a response like
```json
[
    {
        "_id": string,
        "user": {
            "_id": string,
            "firstName": string,
            "lastName": string,
            "email": string,
        },
    },
    // other such request objects
]
```

## Request
Kindly review my claim and let me know the best way forward.

**Comment by palisadoes** [comment]
Please submit a Pull Request so that it can be reviewed by mentors

**Comment by bigBrain1901** [comment]
Sure, thanks!

**Comment by bigBrain1901** [comment]
Sure, thanks!

**Comment by bigBrain1901** [comment]
Hi, I have opened the PR GH-549 with the bugfix and UI-fixes for the Settings page. A description of the same is added to the PR as well.

I can create a separate issue and link it up if needed.  
Please let me know if I have missed some guidelines while doing so.

Once this PR is approved, I will write the test for the file as well :v:

**Comment by palisadoes** [comment]
Unassigning due to inactivity others need to get a chance. Please let us know whether you want to be reassigned before someone else requests assignment or select another issue.

**Comment by bigBrain1901** [comment]
Completion of this issue will help with #721. If noone picks it up, I'll be covering this test on that PR anyways.

**Comment by palisadoes** [comment]
@bigBrain1901 Make this issue auto close with that PR too.

**Comment by palisadoes** [comment]
Unassigning due to inactivity and no submitted PR. Others need to get a chance. Please let us know whether you want to be reassigned before someone else requests assignment or select another issue.

**Comment by Maagztech** [comment]
Please assign me this issue.

**Comment by Sumitmaithani** [comment]
@Maagztech are you working on this issue. If not @palisadoes can I take this issue...

**Comment by Sumitmaithani** [comment]
Great!

**Comment by palisadoes** [comment]
Unassigning due to inactivity and no submitted PR. Others need to get a chance. Please let us know whether you want to be reassigned before someone else requests assignment or select another issue.

**Comment by Sumitmaithani** [comment]
@palisadoes can I take this issue
